### PR TITLE
Chore: use eslint-plugin-rulesdir instead of --rulesdir for self-linting

### PR DESCRIPTION
**Pull request by not-an-aardvark**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Other, please explain:

**What changes did you make? (Give an overview)**

This updates our internal linting build to use `eslint-plugin-rulesdir` rather than the `--rulesdir` CLI flag.

We've been talking about deprecating `rulesdir` for awhile, so I think it would be good for us to dogfood some of the alternatives for using project-specific linting rules.

https://github.com/eslint/eslint/pull/9163 is another way of solving the same problem. This PR is mutually exclusive with https://github.com/eslint/eslint/pull/9163.

**Is there anything you'd like reviewers to focus on?**

We should choose between this approach and the approach in https://github.com/eslint/eslint/pull/9163. Which do you prefer?

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    ],
+    rules: {
+        "eslint-plugin/consistent-output": "error",
+        "eslint-plugin/no-deprecated-context-methods": "error",
+        "eslint-plugin/no-identical-tests": "error",
+        "eslint-plugin/prefer-output-null": "error",
+        "eslint-plugin/prefer-placeholders": "error",
+        "eslint-plugin/report-message-format": ["error", "[^a-z].*\\.$"],
+        "eslint-plugin/test-case-property-ordering": "error",
+        "eslint-plugin/test-case-shorthand-strings": "error"
```
Comment (Aladdin-ADD) [inline .eslintrc.js]: "eslint-plugin/no-identical-tests" has been added in `recommended`.

File: lib/rules/.eslintrc.yml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 rules:
-    internal-no-invalid-meta: "error"
-    internal-consistent-docs-description: "error"
+    rulesdir/no-invalid-meta: "error"
```
Comment (Aladdin-ADD) [inline lib/rules/.eslintrc.yml]: what about change this to `.eslintrc.js` -- for consistency.

File: .eslintrc.js (hunk 1 above)
Comment (not-an-aardvark) [reply .eslintrc.js]: Good catch, but I think we should remove it in a separate PR -- this PR is only supposed to change the format of the config file without modifying how it works.

File: lib/rules/.eslintrc.yml (hunk 2 above)
Comment (not-an-aardvark) [reply lib/rules/.eslintrc.yml]: ¯\\\_(ツ)_/¯ I'm fine with this, but I was assuming people would prefer to keep files as `.yml` wherever possible, and only use `.js` config files when necessary.

File: .eslintrc.js (hunk 1 above)
Comment (Aladdin-ADD) [reply .eslintrc.js]: it has been removed in the master. so I think better not to add it again.

Comment (not-an-aardvark) [reply .eslintrc.js]: Oh, oops. I hadn't realized that it was removed on `master`.

**Pull request merged**